backend/src/users/users.controller.ts

```diff
@@ -40,8 +40,24 @@ export class UsersController {
     }
 
     @Get()
-    async findAllUsers(): Promise<User[]> {
-        return this.usersService.findAll();
+    async findByToken(@Headers('Token') header: string){
+        if(!header) {
+            return this.findAllUsers();
+        }
+        const user = await this.usersService.findByToken(header);
+
+        const { password, ...rest } = user;
+        return rest
+    }
+
+    @Get()
+    async findAllUsers() {
+        const users = await this.usersService.findAll();
+        const filteredUsers = users.map(user => {
```

Comment:
Instead of mapping and filtering out unwanted fields here you can use MongoDB projections to get only desired values from the db: https://docs.mongodb.com/manual/tutorial/project-fields-from-query-results/#return-the-specified-fields-and-the-_id-field-only

You can do it in `user.service.ts`.